### ISSUE TITLE
Bug: OneDrive shows internet connectivity issue even though internet works for everything else

### ISSUE THREAD

**Issue by kabirbg** [abraunegg/onedrive#2127]
### Describe the bug

What it says in the title. I've seen sort of similar things in closed issues, but nothing there seems to work for what is happening to me: I keep getting `ERROR: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?` and everything else loads on my wifi connection. I rebuilt onedrive from the latest source with debugging options, but the attached screenshot is all I'm getting. Not sure what else I could do to troubleshoot as the onedrive logs aren't telling me anything else.

### Operating System Details

```shell
Linux gentoo 5.15.62-gentoo-dist #1 SMP Thu Aug 25 16:57:32 EDT 2022 x86_64 12th Gen Intel(R) Core(TM) i7-12700K GenuineIntel GNU/Linux
```

### Client Installation Method

From Source

### OneDrive Account Type

Personal

### What is your OneDrive Application Version

onedrive v2.4.20

### What is your OneDrive Application Configuration

```shell
onedrive version                             = v2.4.20
Config path                                  = /home/kabir/.config/onedrive
Config file found in config path             = false
Config option 'sync_dir'                     = /home/kabir/OneDrive
Config option 'enable_logging'               = false
Config option 'log_dir'                      = /var/log/onedrive/
Config option 'disable_notifications'        = false
Config option 'min_notify_changes'           = 5
Config option 'skip_dir'                     =
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 300
Config option 'monitor_log_frequency'        = 5
Config option 'monitor_fullscan_frequency'   = 12
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = false
Config option 'check_nomount'                = false
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'space_reservation'            = 52428800
Config option 'application_id'               =
Config option 'azure_ad_endpoint'            =
Config option 'azure_tenant_id'              = common
Config option 'user_agent'                   =
Config option 'force_http_11'                = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'sync_root_files'              = false
Selective sync 'sync_list' configured        = false
Config option 'sync_business_shared_folders' = false
Business Shared Folders configured           = false
Config option 'webhook_enabled'              = false
```

### What is your 'curl' version

```shell
curl 7.84.0 (x86_64-pc-linux-gnu) libcurl/7.84.0 OpenSSL/1.1.1q zlib/1.2.12 nghttp2/1.47.0
Release-Date: 2022-06-27
Protocols: dict file ftp ftps http https imap imaps mqtt pop3 pop3s rtsp smtp smtps tftp
Features: HTTP2 HTTPS-proxy IPv6 Largefile libz NTLM SSL threadsafe TLS-SRP UnixSockets
```

### Where is your 'sync_dir' located

Local

### What are all your system 'mount points'

```shell
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
sysfs on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
devtmpfs on /dev type devtmpfs (rw,nosuid,size=10240k,nr_inodes=2014143,mode=755,inode64)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev,noexec,inode64)
tmpfs on /run type tmpfs (rw,nosuid,nodev,noexec,mode=755,inode64)
/dev/nvme0n1p3 on / type btrfs (rw,noatime,ssd,space_cache=v2,subvolid=5,subvol=/)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
debugfs on /sys/kernel/debug type debugfs (rw,nosuid,nodev,noexec,relatime)
configfs on /sys/kernel/config type configfs (rw,nosuid,nodev,noexec,relatime)
pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
efivarfs on /sys/firmware/efi/efivars type efivarfs (rw,nosuid,nodev,noexec,relatime)
cgroup_root on /sys/fs/cgroup type tmpfs (rw,nosuid,nodev,noexec,relatime,size=10240k,mode=755,inode64)
openrc on /sys/fs/cgroup/openrc type cgroup (rw,nosuid,nodev,noexec,relatime,release_agent=/lib/rc/sh/cgroup-release-agent.sh,name=openrc)
none on /sys/fs/cgroup/unified type cgroup2 (rw,nosuid,nodev,noexec,relatime,nsdelegate)
cpuset on /sys/fs/cgroup/cpuset type cgroup (rw,nosuid,nodev,noexec,relatime,cpuset)
cpu on /sys/fs/cgroup/cpu type cgroup (rw,nosuid,nodev,noexec,relatime,cpu)
cpuacct on /sys/fs/cgroup/cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpuacct)
blkio on /sys/fs/cgroup/blkio type cgroup (rw,nosuid,nodev,noexec,relatime,blkio)
memory on /sys/fs/cgroup/memory type cgroup (rw,nosuid,nodev,noexec,relatime,memory)
devices on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
freezer on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
net_cls on /sys/fs/cgroup/net_cls type cgroup (rw,nosuid,nodev,noexec,relatime,net_cls)
perf_event on /sys/fs/cgroup/perf_event type cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)
net_prio on /sys/fs/cgroup/net_prio type cgroup (rw,nosuid,nodev,noexec,relatime,net_prio)
hugetlb on /sys/fs/cgroup/hugetlb type cgroup (rw,nosuid,nodev,noexec,relatime,hugetlb)
pids on /sys/fs/cgroup/pids type cgroup (rw,nosuid,nodev,noexec,relatime,pids)
misc on /sys/fs/cgroup/misc type cgroup (rw,nosuid,nodev,noexec,relatime,misc)
mqueue on /dev/mqueue type mqueue (rw,nosuid,nodev,noexec,relatime)
binfmt_misc on /proc/sys/fs/binfmt_misc type binfmt_misc (rw,nosuid,nodev,noexec,relatime)
/dev/nvme0n1p1 on /boot type vfat (rw,noatime,fmask=0022,dmask=0022,codepage=437,iocharset=ascii,shortname=mixed,errors=remount-ro)
/dev/sda1 on /mnt/backup type btrfs (rw,nosuid,nodev,noexec,relatime,space_cache=v2,subvolid=5,subvol=/,user)
/dev/sda2 on /mnt/storage type ext4 (rw,nosuid,nodev,noexec,relatime,user)
/dev/nvme0n1p6 on /mnt/windows type fuseblk (ro,relatime,user_id=0,group_id=0,allow_other,blksize=4096)
/dev/sda3 on /mnt/win_files type fuseblk (ro,nosuid,nodev,noexec,relatime,user_id=0,group_id=0,allow_other,blksize=4096,user)
tmpfs on /run/user/1000 type tmpfs (rw,nosuid,nodev,relatime,size=1615620k,nr_inodes=403905,mode=700,uid=1000,gid=1000,inode64)
```

### What are all your local file system partition types

```shell
/ is btrfs, I have some other partitions on ext4 and one Windows NTFS.
```

### How do you use 'onedrive'

I only have onedrive running on my one computer. I use it sometimes to access files on my phone. I use onedrive_tray too, but that's about it.

### Steps to reproduce the behaviour

Just run onedrive --monitor or onedrive --synchronize.

### Complete Verbose Log Output

```shell
Using 'user' Config Dir: /home/kabir/.config/onedrive
Using 'system' Config Dir: /etc/onedrive
No user or system config file found, using application defaults
Checking Application Version ...
Initializing the OneDrive API ...
Configuring Global Azure AD Endpoints
Using Curl defaults for all HTTP operations
Opening the item database ...
All operations will be performed in: /home/kabir/OneDrive

ERROR: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?
  Retry attempt:          1
  This attempt timestamp: 2022-Sep-09 10:06:57
  Next retry in approx:   1 seconds
  Next retry approx:      2022-Sep-09 10:06:58
^C
gentoo git sources/onedrive ‹master› » onedrive --synchronize -vv                                             130 ↵
[DEBUG] homePath: HOME environment variable set
[DEBUG] homePath: /home/kabir
[DEBUG] configDirBase: WARNING - no XDG_CONFIG_HOME environment variable set
[DEBUG] configDirBase: /home/kabir/.config
[DEBUG] configDirName: Configuring application to use default config path
Using 'user' Config Dir: /home/kabir/.config/onedrive
Using 'system' Config Dir: /etc/onedrive
[DEBUG] refreshTokenFilePath = /home/kabir/.config/onedrive/refresh_token
[DEBUG] deltaLinkFilePath = /home/kabir/.config/onedrive/delta_link
[DEBUG] databaseFilePath = /home/kabir/.config/onedrive/items.sqlite3
[DEBUG] databaseFilePathDryRun = /home/kabir/.config/onedrive/items-dryrun.sqlite3
[DEBUG] uploadStateFilePath = /home/kabir/.config/onedrive/resume_upload
[DEBUG] userConfigFilePath = /home/kabir/.config/onedrive/config
[DEBUG] syncListFilePath = /home/kabir/.config/onedrive/sync_list
[DEBUG] systemConfigFilePath = /etc/onedrive/config
[DEBUG] businessSharedFolderFilePath = /home/kabir/.config/onedrive/business_shared_folders
No user or system config file found, using application defaults
[DEBUG] passed in options: ["onedrive", "--synchronize"]
[DEBUG] note --confdir and --verbose not listed in args
[DEBUG] sync_dir: Getting syncDir from config value sync_dir
[DEBUG] sync_dir: A '~' was found in configured sync_dir, automatically expanding as SHELL and USER environment variable is set
[DEBUG] syncDir: /home/kabir/OneDrive
[DEBUG] Testing network to ensure network connectivity to Microsoft OneDrive Service
[DEBUG] Attempting to contact online service
[DEBUG] Shutting down HTTP engine as successfully reached OneDrive Online Service
Checking Application Version ...
Initializing the OneDrive API ...
Configuring Global Azure AD Endpoints
[DEBUG] Configured authUrl:          https://login.microsoftonline.com/common/oauth2/v2.0/authorize
[DEBUG] Configured redirectUrl:      https://login.microsoftonline.com/common/oauth2/nativeclient
[DEBUG] Configured tokenUrl:         https://login.microsoftonline.com/common/oauth2/v2.0/token
[DEBUG] Configured driveUrl:         https://graph.microsoft.com/v1.0/me/drive
[DEBUG] Configured driveByIdUrl:     https://graph.microsoft.com/v1.0/drives/
[DEBUG] Configured sharedWithMeUrl:  https://graph.microsoft.com/v1.0/me/drive/sharedWithMe
[DEBUG] Configured itemByIdUrl:      https://graph.microsoft.com/v1.0/me/drive/items/
[DEBUG] Configured itemByPathUrl:    https://graph.microsoft.com/v1.0/me/drive/root:/
[DEBUG] Configured siteSearchUrl:    https://graph.microsoft.com/v1.0/sites?search
[DEBUG] Configured siteDriveUrl:     https://graph.microsoft.com/v1.0/sites/
Using Curl defaults for all HTTP operations
[DEBUG] clientId    = d50ca740-c83f-4d1b-b616-12c519384f0c
[DEBUG] companyName = abraunegg
[DEBUG] appTitle    = OneDrive Client for Linux
Opening the item database ...
[DEBUG] Using database file: /home/kabir/.config/onedrive/items.sqlite3
[DEBUG] Configuring default new folder permissions as: 700
[DEBUG] Configuring default new file permissions as: 600
All operations will be performed in: /home/kabir/OneDrive
[DEBUG] Configuring skip_dir ...
[DEBUG] skip_dir:
[DEBUG] Configuring skip_dir_strict_match ...
[DEBUG] skip_dir_strict_match: false
[DEBUG] Configuring skip_dotfiles ...
[DEBUG] skip_dotfiles: false
[DEBUG] Configuring skip_file ...
[DEBUG] skip_file: ~*|.~*|*.tmp
[DEBUG] Setting file permissions for: /home/kabir/.config/onedrive/refresh_token
[DEBUG] Request URL = https://graph.microsoft.com/v1.0/me/drive
[DEBUG] onedrive.perform() Generated a OneDrive CurlException
[DEBUG] A timeout was triggered - data too slow, no response ... use --debug-https to diagnose further
[DEBUG] Configuring libcurl to use a fresh connection for re-try

ERROR: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?
  Retry attempt:          1
  This attempt timestamp: 2022-Sep-09 10:07:21
  Next retry in approx:   1 seconds
  Next retry approx:      2022-Sep-09 10:07:22
[DEBUG] Configuring libcurl to use a fresh connection for re-try

ERROR: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?
  Retry attempt:          2
  This attempt timestamp: 2022-Sep-09 10:07:32
  Next retry in approx:   4 seconds
  Next retry approx:      2022-Sep-09 10:07:36
[DEBUG] Configuring libcurl to use a fresh connection for re-try

ERROR: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?
  Retry attempt:          3
  This attempt timestamp: 2022-Sep-09 10:07:46
  Next retry in approx:   9 seconds
  Next retry approx:      2022-Sep-09 10:07:55
[DEBUG] Configuring libcurl to use a fresh connection for re-try
```

### Screenshots

![image](https://user-images.githubusercontent.com/56413904/189368334-33ac21a6-55f7-4956-977f-a9a2261fd692.png)

**Comment by abraunegg** [comment]
@kabirbg 
This is *not* a bug with this application - this is an issue with:
* Your local host
* Your DNS service
* Your router
* Your firewall
* Your ISP

Please read: https://github.com/abraunegg/onedrive/wiki/Frequently-Asked-Questions#cannot-connect-to-microsoft-onedrive-service---network-connection-issue

**Comment by kabirbg** [comment]
@abraunegg Thank you. Yeah, because onedrive has been working perfectly until now I didn't think it was a bug, but couldn't figure out what I was doing wrong. The curl command on your FAQ page works perfectly for me and the connection loads. I can access OneDrive and all other websites from my browser, and so I don't know what else to try...

**Comment by abraunegg** [comment]
@kabirbg 
You *will* have something blocking the client from going outbound from your system.

The binary 'onedrive' makes calls to 'curl' for curl to make the queries to OneDrive.

That no message was additionally printed regarding what triggered this, means that it is the 'curl' process itself that is having the issue.

> I can access OneDrive and all other websites from my browser

This is actually a rather pointless test ... because those do not use the same process ...

> .. I don't know what else to try...

There has to be something done locally with your system where the client now has some sort of issue.

Please try this command .. it will be VERY verbose, but this is also debugging the actual HTTPS connection going outbound using curl

Please **DO NOT** post any output from this command as it will contain all your highly sensitive account information:

```
onedrive --synchronize --verbose --verbose --debug-https
```

You will need to go through that output and see where any why curl is having an issue.

**Comment by kabirbg** [comment]
Hi, thank you so much for your detailed answer.
I tried some more debugging but am at a loss, as it appears to log in just fine but there's a hangup in trying to connect to https://graph.microsoft.com/v1.0/me/drive. Testing graph.microsoft.com with curl also gives me `Failed to connect: Connection timed out`. But now I am at a loss as to what to do with that information. Is it just that the microsoft website is down right now?? 
Sorry for my inexperience and for bothering you with this but was not sure where else to ask for help.

**Comment by abraunegg** [comment]
@kabirbg 
You are blocking https connections to graph.microsoft.com somehow somewhere

Check your local firewall setup, check your router ... something in your environment is the cause.

This issue will be something entirely local to you.

**Comment by kabirbg** [comment]
Hey abraunegg, thank you so so much for taking the time to help me. I fished around in my router settings and it turns out that gaming sites were blocked on my router, which for some reason included Microsoft (Xbox, but why _all_ microsoft services?? beats me). I fixed that, and it's working like a charm now. I'm so sorry to have taken up your time, and thank you for your careful help! And for developing this awesome client, it has been such a lifesaver and I've used it for 3+ years now!

**Comment by abraunegg** [comment]
@kabirbg 
Glad you got your issue sorted. Will update the FAQ to include to do the curl tests against https://graph.microsoft.com as well given your experience.

> And for developing this awesome client, it has been such a lifesaver and I've used it for 3+ years now!

Thanks for the feedback, greatly appreciated.

**Comment by abraunegg** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
